web3js-ext/test/role_based_key_feepayer.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { Web3, TxType, AccountKeyType, getPublicKeyFromPrivate, toPeb } from "@kaiachain/web3js-ext";
+import { assert } from "chai";
+
+const senderAddr = "0x334b4d3c775c45c59de54e9f0408cba25a1aece7";
+const senderRoleTransactionPriv = "0xc9668ccd35fc20587aa37a48838b48ccc13cf14dd74c8999dd6a480212d5f7ac";
+const senderRoleAccountUpdatePriv = "0x9ba8cb8f60044058a9e6f815c5c42d3a216f47044c61a1750b6d29ddc7f34bda";
+const senderRoleFeePayerPriv = "0x0e4ca6d38096ad99324de0dde108587e5d7c600165ae4cd6a2462c597458c2b8";
+const receiverAddr = "0xc40b6909eb7085590e1c26cb3becc25368e249e9";
+
+const provider = new Web3.providers.HttpProvider("https://public-en-kairos.node.kaia.io");
+const web3 = new Web3(provider);
+
+const roleAccountUpdate = web3.eth.accounts.privateKeyToAccount(senderRoleAccountUpdatePriv);
+const roleTransactionAccount = web3.eth.accounts.privateKeyToAccount(senderRoleTransactionPriv);
+const roleFeePayerAccount = web3.eth.accounts.privateKeyToAccount(senderRoleFeePayerPriv);
+
+async function checkBalance(address: string) {
+    const balance = await web3.eth.getBalance(address);
+    console.log(`Balance of ${address}: ${web3.utils.fromWei(balance, "ether")} KLAY`);
+    return parseFloat(web3.utils.fromWei(balance, "ether"));
+}
+
+describe("Role-based Key Tests", function () {
+    this.timeout(10000);
+
+    // Before all tests, set up Role-based Key
+    before(async function () {
+        console.log("\n--- Setting Role-based Key ---");
+        const pub1 = getPublicKeyFromPrivate(senderRoleTransactionPriv);
+        const pub2 = getPublicKeyFromPrivate(senderRoleAccountUpdatePriv);
+        const pub3 = getPublicKeyFromPrivate(senderRoleFeePayerPriv);
+
+        const updateTx = {
+            type: TxType.AccountUpdate,
+            from: senderAddr,
+            gasLimit: 100000,
+            key: {
+                type: AccountKeyType.RoleBased,
+                keys: [
+                    { type: AccountKeyType.Public, key: pub1 },
+                    { type: AccountKeyType.Public, key: pub2 },
+                    { type: AccountKeyType.Public, key: pub3 }
+                ]
+            }
+        };
+
+        const signedUpdateTx = await roleAccountUpdate.signTransaction(updateTx);
```

Comment:
@minminkikiki I think this unittest is not general because the private keys are public. That means it will be failed when someone changed the accountUpdateKey.

Could you fix this test to generate temporary key and update the account key to role based?

Reply:
And we should remove the dependencies with network connection. So we don't need to send the tx to the testnet. We can just check the signed result.